### PR TITLE
Main

### PR DESCRIPTION
**Pull request by OlegMR0**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 .idea/*
 *.iml
 target/*
+src\main\java\mate\jdbc\PasswordDao.java
```
Comment (okuzan) [inline .gitignore:4]: undo this change

File: src/main/java/mate/jdbc/Main.java [hunk 2]
```diff
@@ -1,7 +1,21 @@
...
+    public static void main(String[] args) {
+        ManufacturerDao manufacturerDao = (ManufacturerDao) injector
+                .getInstance(ManufacturerDao.class);
+        Manufacturer manufacturer1 = manufacturerDao
+                .create(new Manufacturer("Name123", "Country123"));
+        manufacturerDao.delete(3L);
+        manufacturerDao.get(Long.valueOf(1));
+        manufacturer1.setName("NewName");
+        manufacturerDao.update(manufacturer1);
+        System.out.println(manufacturerDao.getAll());
```
Comment (okuzan) [inline src/main/java/mate/jdbc/Main.java:19]: naming with numbers is a bad practice, don't use hardcoded ids here, use getId() of existing entity objects

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.DataBaseConnector;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+
```
Comment (okuzan) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:17]: empty line

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create manufacturer!", e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        try (Connection connection = DataBaseConnector.getConnection(); PreparedStatement statement
```
Comment (okuzan) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:41]:
```suggestion
        try (Connection connection = DataBaseConnector.getConnection(); 
```
let place each resource on the new line

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            ResultSet resultSet = statement
+                    .executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getManufacturerFromResultSet(resultSet));
+            }
+            throw new DataProcessingException("Can't find manufacturer!", new Throwable());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find manufacturer!", e);
+        }
+    }
```
Comment (okuzan) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:52]:
```suggestion
        } catch (SQLException e) {
            throw new DataProcessingException("Can't find manufacturer!", e);
        }
        return Optional.empty();
    }
}
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    public Manufacturer update(Manufacturer manufacturer) {
+        try (Connection connection = DataBaseConnector.getConnection(); PreparedStatement
+                statement = connection.prepareStatement(Query.UPDATE.string)) {
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.setLong(3, manufacturer.getId());
+            statement.executeUpdate();
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer!", e);
```
Comment (okuzan) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:79]: include manufacturer or at least its id in the message

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        try (Connection connection = DataBaseConnector.getConnection(); PreparedStatement
+                preparedStatement = connection.prepareStatement(Query.DELETE.string)) {
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
```
Comment (okuzan) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:90]: follow the same pattern for exception handling as in other methods

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+        DELETE("UPDATE manufacturers SET is_deleted = TRUE WHERE id = ?");
+
+        private String string;
+
+        Query(String string) {
+            this.string = string;
+        }
+    }
+}
+
```
Comment (okuzan) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:116]: empty line

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            } else {
+                throw new DataProcessingException("Insert error!", new Throwable());
+            }
```
Comment (okuzan) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:32]:
```suggestion
            } 
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.DataBaseConnector;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        try (Connection connection = DataBaseConnector.getConnection(); PreparedStatement
+                statement = connection.prepareStatement((Query.INSERT.string),
```
Comment (okuzan) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:21]:
```suggestion
                statement = connection.prepareStatement(Query.INSERT.string,

```

**Review by okuzan — CHANGES_REQUESTED** [review]
For each comment, make sure you fixed the pointed-out issue everywhere

File: .gitignore (hunk 1 above)
Comment (OlegMR0) [reply .gitignore:4]: But why? I have stored there my password, it's just convenient for me.

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java (hunk 4 above)
Comment (OlegMR0) [reply src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:41]: Maven checkstyle demands extra whitespaces, that don't offered by IDEA, so it looks strange to me.

File: .gitignore (hunk 1 above)
Comment (kozhukhovsky) [reply .gitignore]: Passwords are not stored that way. Thus, if I cloned the application, I would not be able to enter my password in any way to gain access to the database. I would need to create a PasswordDao.
Usually done or like you did with ConnectionUtil. You can also add a property file, but you'll get to know this later on in the course.

File: src/main/java/mate/jdbc/util/DataBaseConnector.java [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+        try {
+            return DriverManager
+                    .getConnection("jdbc:mysql://localhost:3306/taxi-service", USER, PASSWORD);
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't get connection", e);
+        }
+    }
+}
+
+
```
Comment (kozhukhovsky) [inline src/main/java/mate/jdbc/util/DataBaseConnector.java:29]:
```suggestion
```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+        try (Connection connection = DataBaseConnector.getConnection();
+                  PreparedStatement statement = connection.prepareStatement(Query.UPDATE.string)) {
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.setLong(3, manufacturer.getId());
+            statement.executeUpdate();
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update manufacturer!"
+                    + manufacturer.toString(), e);
```
Comment (kozhukhovsky) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:80]:
```suggestion
                    + manufacturer, e);

```

File: src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+            statement.setString(2, manufacturer.getCountry());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                manufacturer.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create manufacturer!"
+                    + manufacturer.toString(), e);
```
Comment (kozhukhovsky) [inline src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java:32]:
```suggestion
                    + manufacturer,  e);

```

**Review by kozhukhovsky — APPROVED** [review]
